backend/controller/ingress/ingress.go

```diff
@@ -249,3 +254,46 @@ func validateValue(fieldType schema.Type, path path, value any, sch *schema.Sche
 	}
 	return nil
 }
+
+func parseQueryParams(values url.Values) (map[string]any, error) {
+	if jsonStr, ok := values["json"]; ok {
+		if len(jsonStr) > 1 {
+			return nil, fmt.Errorf("multiple JSON query parameters provided")
+		}
+
+		return decodeQueryJSON(jsonStr[0])
+	}
+
+	queryMap := make(map[string]any)
+	for key, value := range values {
+		if len(value) == 1 {
+			queryMap[key] = value[0]
+		} else {
+			for _, v := range value {
+				if strings.Contains(v, "[") || strings.Contains(v, "{") {
+					return nil, fmt.Errorf("complex structures like nested arrays/maps are not supported: [%s], use 'json=' instead", key)
```

Comment:
I don't think this is a terrible compromise, but  think we should support field references with `.`.

Also shouldn't this check be against the key as well, and against `value[0]` in the single-element case?